tileserver/__init__.py

```diff
@@ -273,12 +298,19 @@ def handle_request(self, request):
                 tile_data = reformat_selected_layers(
                     tile_data_all, self.layer_config.all_layers, coord, format)
 
-                # note that we want to store the data too, as this means that
-                # future requests can be serviced directly from the store.
+                # note that we want to store the formatted data too,
+                # as this means that future requests can be serviced
+                # directly from the store.
                 if self.store and coord.zoom <= 20:
                     self.io_pool.apply_async(
                         async_store, (self.store, tile_data, coord, format))
 
+            # since we're a request for all, enqueue the coordinate to
+            # ensure other formats get processed
+            if self.sqs_queue and coord.zoom <= 20:
```

Comment:
Yes that's right. But if we always enqueued, the scenario that wouldn't work out so well would be consistently receiving requests for a custom layer, eg `buildings`. We would enqueue tiles for every request, even though nothing would be changing.

I think it makes sense to enqueue when we receive a request for something that would have gotten served by the cache. Currently it's just the `all` layer, but we talked about starting to store tiles for the popular layers like `buildings`. If we did, then we would want to enqueue and store when we receive requests for these as well.

I'm inclined to leave it the way it is currently, and when code gets added to start storing other layers on s3, update this bit as well to reflect that, maybe through some explicit configuration in case we want to add more combinations in the future.

Thoughts?